odbc/src/api/get_type_info_tests.rs

```diff
@@ -0,0 +1,216 @@
+            "bool",
+            "long",
+            "binData",
+            "array",
+            "bson",
+            "dbPointer",
+            "decimal",
+            "javascript",
+            "javascriptWithScope",
+            "maxKey",
+            "minKey",
+            "null",
+            "object",
+            "objectId",
+            "symbol",
+            "timestamp",
+            "undefined",
+            "int",
+            "double",
+            "string",
+            "date",
+        ];
+        validate_result_set(SqlDataType::UNKNOWN_TYPE, expectations);
+    }
+
+    #[test]
+    fn test_specific_type_one_response() {
+        let expectations = vec!["int"];
+        validate_result_set(SqlDataType::INTEGER, expectations);
+    }
+
+    #[test]
+    fn test_invalid_type_error() {
+        // Test that a sql data type that is not defined in the enum yields the correct error
+        let handle: *mut _ = &mut MongoHandle::Statement(Statement::with_state(
+            std::ptr::null_mut(),
+            StatementState::Allocated,
+        ));
+        unsafe {
+            assert_eq!(SqlReturn::ERROR, SQLGetTypeInfo(handle as *mut _, 100));
+            // use SQLGetDiagField to retreive and assert correct error message
+            let message_text = &mut [0u16; 6] as *mut _ as *mut c_void;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetDiagFieldW(
+                    Stmt,
+                    handle as *mut _,
+                    1,
+                    DiagType::SQL_DIAG_SQLSTATE as i16,
+                    message_text,
+                    6,
+                    &mut 0
+                )
+            );
+            assert_eq!(
+                INVALID_SQL_TYPE,
+                String::from_utf16(&*(message_text as *const [u16; 6])).unwrap()
+            );
+        }
+    }
+
+    #[test]
+    fn test_unsupported_type_empty_list() {
+        // given a valid type defined in the enum, but not supported, yield an empty list
+        validate_result_set(SqlDataType::INTERVAL_HOUR_TO_MINUTE, vec![]);
+    }
+
+    #[test]
+    fn test_invalid_cursor_state_error() {
+        // checks for invalid cursor state when calling get_value before next
+        let handle: *mut _ = &mut MongoHandle::Statement(Statement::with_state(
+            std::ptr::null_mut(),
+            StatementState::Allocated,
+        ));
+        unsafe {
+            let stmt = (*handle).as_statement().unwrap();
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetTypeInfo(handle as *mut _, SqlDataType::INTEGER as i16)
+            );
+            let value = stmt
+                .mongo_statement
+                .write()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get_value(1);
+            assert!(value.is_err());
+        }
+    }
+
+    #[test]
+    fn test_get_all_type_info_row_values() {
```

Comment:
Before submitting can you move this, test_unsupported_type_empty_list and test_specific_type_one_response into function.yaml, now that we have everything in place to utilize the test harness?
Please don't hate me, this is the last change needed and don't need to retag me before merging.